Sources/StytchUI/StytchB2BUIClient/Managers/OrganizationManager.swift

```diff
@@ -0,0 +1,61 @@
+import StytchCore
+
+struct OrganizationManager {
+    private static var searchedOrganization: StytchB2BClient.SearchManager.OrganizationSearchResponse?
+
+    static var allowedAuthMethods: [StytchB2BClient.AllowedAuthMethods]? {
+        if let organization = StytchB2BClient.organizations.getSync() {
```

Comment:
I don't know if I love this. Relying on/using state outside of the UI's immediate control _feels_ like it could cause problems. Like, off the top of my head, if a user is authenticated with one organization (so `StytchB2BClient.organizations.getSync()` returns `organzationA`) , and then starts an auth flow for a _new_ organization (so `searchedOrganization` is `organizationB`), it would cause issues, no? I don't actually know if that's possible, but bringing it up

Reply:
Ok lets discuss I can easily separate it

Reply:
Ok I refactored this to make a new protocol called `OrganizationType` that both the return value from the search by slug and the regular Organization conform to, therefore making it opaque which kind we are dealing with. This will make it easier in the newly created `OrganizationManager` going forward.